AK/StringFloatingPointConversions.cpp

```diff
@@ -0,0 +1,1126 @@
+
+        // Checks if value is divisible by 5 ^ power.
+        auto is_divisible_by_pow_5 = [](u64 value, i32 power) {
+            constexpr Array<u64, 5> powers_of_five = { { 5, 25, 625, 390625, 152587890625 } };
+
+            if (power <= 0 || value == 0)
+                return true;
+            if (power >= 28) // 2 ^ 64 - 1 < 5 ^ 28
+                return false;
+
+            i32 result = 0;
+            for (i32 i = 5; i--;) {
+                if (value % powers_of_five[i] == 0) {
+                    value /= powers_of_five[i];
+                    result += 1 << i;
+                }
+            }
+            return result >= power;
+        };
+
+        all_u_zero = is_divisible_by_pow_5(u, skipped_iters);
+        all_v_zero = is_divisible_by_pow_5(v, skipped_iters);
+        all_w_zero = is_divisible_by_pow_5(w, skipped_iters);
+
+        auto multiplier = lookup_table<FloatingPoint>.ge[skipped_iters];
+        i32 k = log2_5_num * skipped_iters / log2_5_denum + lookup_table<FloatingPoint>.b1;
+        u = multiply_and_shift(u, multiplier, skipped_iters + k - synthetic_exponent);
+        v = multiply_and_shift(v, multiplier, skipped_iters + k - synthetic_exponent);
+        w = multiply_and_shift(w, multiplier, skipped_iters + k - synthetic_exponent);
+    }
+
+    // Step 4'. Find the shortest, correctly-rounded decimal representation in the interval.
+    bool is_even = ~mantissa & 1;
+    bool accept_smaller = is_even && all_u_zero;
+    bool accept_larger = is_even || !all_w_zero;
+
+    if (!accept_larger)
+        --w;
+
+    bool all_a_zero = accept_smaller;
+    bool all_b_zero = all_v_zero;
+    int last_digit = 0;
+
+    int exponent10 = skipped_iters - max(-synthetic_exponent, 0);
+
+    while (u / 10 < w / 10) {
+        all_a_zero &= u % 10 == 0;
+        all_b_zero &= last_digit == 0;
+        last_digit = v % 10;
+
+        u /= 10;
+        v /= 10;
+        w /= 10;
+        ++exponent10;
+    }
+    if (all_a_zero) {
+        while (u % 10 == 0) {
+            all_b_zero &= last_digit == 0;
+            last_digit = v % 10;
+
+            u /= 10;
+            v /= 10;
+            w /= 10;
+            ++exponent10;
+        }
+    }
+
+    bool is_tie = all_b_zero && last_digit == 5;
+    bool want_round_down = last_digit < 5 || (is_tie && v % 2 == 0);
+    bool round_down = (want_round_down && (u != v || all_a_zero)) || (v + 1 > w);
+    return { sign, round_down ? v : v + 1, exponent10 };
+}
+
+static u128 multiply(u64 a, u64 b)
+{
+#ifdef __SIZEOF_INT128__
+    unsigned __int128 result = (unsigned __int128)a * b;
+    u64 low = result;
+    u64 high = result >> 64;
+    return u128 { low, high };
+#else
+    return u128 { a }.wide_multiply(u128 { b }).low;
+#endif
+}
+
+template<>
+FloatingPointExponentialForm convert_floating_point_to_decimal_exponential_form<float>(float value)
+{
+    auto multiply_and_shift = [&](u64 operand, u64 multiplier, i32 shift) {
```

Comment:
Since this lambda doesn't use `value` you can just use an empty capture group `[](u64 operand, ...`